iosxr_test.py

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/python
+'''
+Author: Rich Wellum (richwellum@gmail.com)
+
+A tool to run some basic tests on a IOS XRv vagrant box, using pxssh module.
+
+This will verify both IOS XR Linux and IOS XR Console access.
+
+Is called from iosxr_iso2vbox post generation to verify sanity of
+the Virtualbox,
+
+Can also be run manually, like this:
+
+python iosxr_test.py iosxrv-fullk9-x64.box
+'''
+
+from __future__ import print_function
+import sys
+import pexpect
+from pexpect import pxssh
+import subprocess
+import argparse
+import os
+from iosxr_iso2vbox import run
+
+try:
+    raw_input
+except NameError:
+    raw_input = input
+
+# Some defaults
+terminal_type = 'ansi'
+linux_prompt = r"[#$]"
+xr_prompt = "[$#]"
+login_timeout = 10
+hostname = "localhost"
+username = "vagrant"
+password = "vagrant"
+linux_port = 2222
+xr_port = 2223
+
+
+def bringup_vagrant(input_box):
+    '''
+    Bring up a vagrant box.
+
+    Clean up ssh keys from the generation of the virtualbox.
+
+    Use pxssh to fascilitate the ssh process.
+    '''
+    # Clean up Vagrantfile
+    try:
+        os.remove('Vagrantfile')
+    except OSError:
+        pass
+
+    print('Cleaning up ssh keys')
+    run('ssh-keygen -R [localhost]:2222')
+    run('ssh-keygen -R [localhost]:2223')
+
+    print("Bringing up '%s'..." % input_box)
+
+    run('vagrant init XRv64')  # Single node for now, in future could bring up two nodes and do more testing
```

Comment:
Ack to the first.

I think for the second, that it's implicit that the user is generating boxes, and it's a different machine to what they are running on. Because iso2vbox cleans up any related running VM's already. To do this I'd move the input Box to a temp directory, run the testing and move the box back - but I'm not sure this gains too much with the current design.